test/Faker/Provider/PersonTest.php

```diff
@@ -84,4 +86,42 @@ public function testNameReturnsFirstNameAndLastName()
         $this->assertContains($faker->name('male'), array('John Doe'));
         $this->assertContains($faker->name('female'), array('Jane Doe'));
     }
+
+    public function localeProviderForPerson()
+    {
+        $locales = array();
+        foreach (glob(__DIR__ . '/../../../src/Faker/Provider/*/Person.php') as $localizedPerson) {
```

Comment:
Im not happy with these kind of tests to parse php files. @localheinz what do you think?

Reply:
@ktzouno could you update those tests and create something that does not scan the actual code but just consumes it?

Reply:
@pimjansen I see similar implementations using glob() in `InternetTest::localeDataProvider()`, `PaymentTest::localeDataProvider()` and `ProviderOverrideTest::localeDataProvider()`. Is your point still about the phpunit-provider or the test itself?

Reply:
Yeah but its not really nice to do it. So please update accordingly. Would be great if we can make it asap. Else we have to remove it from the 1.9 release

Reply:
Is it better if I get the list of locales from `Faker\Provider\Miscellaneous::$localeData`? Of course not all of them will have a `Person` provider.

Reply:
Well what i would so is remove the name `Facebook` from the code since it has nothing to do with Facebook in general, it is just an use case. Further throw an exception when a name with 3 consecutive characters pops up. In the test just feed the method where you expect if the exception is thrown once it happens.

Reply:
> Well what i would so is remove the name Facebook from the code since it has nothing to do with Facebook in general, it is just an use case.

Sounds fine.

> In the test just feed the method where you expect if the exception is thrown once it happens.

I don't get this part. (What I'd like to test is that no `firstNameMale`, `firstNameFemale`, `lastName` entries in all locales' `Person` Providers contain 3+ consecutive-identical characters).

Reply:
@ktzouno @pimjansen 

Perhaps something like this makes sense instead?

```php
<?php

namespace Faker\Test\Provider;

use Faker\Generator;
use Faker\Provider\Person;
use PHPUnit\Framework\TestCase;

abstract class AbstractPersonTestCase extends TestCase
{
    /**
     * @dataProvider facebookCompliantPropertyNameProvider
     *
     * @param string $propertyName
     */
    final public function testMagicPropertyProvidesFacebookCompliantName($propertyName)
    {
        $provider = static::fakerWithAddedPersonProvider();

        self::assertValueProvidedByMagicPropertyIsFacebookCompliantName($propertyName, $provider->{$propertyName});

    }

    final public function facebookCompliantPropertyNameProvider()
    {
        return array(
            'firstNameMale' => array('firstNameMale'),
            'firstNameFemale' => array('firstNameFemale'),
            'lastName' => array('lastName'),
        );
    }

    /**
     * @param string $propertyName
     * @param string $value
     */
    final protected static function assertValueProvidedByMagicPropertyIsFacebookCompliantName($propertyName, $value)
    {
        self::assertInternalType('string', $propertyName);
        self::assertInternalType('string', $value);

        self::assertNotRegExp('/(.)\\1{2,}/', $propertyName, sprintf(
            'Failed asserting that the "%s" provider returned a name that is compliant with Facebook, as "%s" has more than 3 consecutive identical characters.',
            $propertyName,
            $value
        ));
    }

    /**
     * @throws \RuntimeException
     *
     * @return Generator
     */
    final protected static function fakerWithAddedPersonProvider()
    {
        $className = self::personProviderClassName();

        if (!class_exists($className)) {
            throw new \RuntimeException(sprintf(
                'Class "%s", resolved from test class name "%s", does not exist',
                $className,
                get_called_class()
            ));
        }

        $faker = new Generator();

        $provider = new $className($faker);

        if (!$provider instanceof Person) {
            throw new \RuntimeException(sprintf(
                'Class "%s" does not extend "Faker\\Provider\\Person"',
                $className
            ));
        }

        $faker->addProvider($provider);

        return $faker;
    }

    /**
     * @return string
     */
    private static function personProviderClassName()
    {
        return preg_replace(
            '/Test$/',
            '',
            str_replace(
                'Faker\\Test\\Provider\\',
                'Faker\\Provider\\',
                get_called_class()
            )
        );
    }
}
```

The individual `PersonTest` classes could then extend from this `AbstractTestCase`.

A bit less magic, perhaps?